lib/inspec/dsl.rb

```diff
@@ -31,6 +31,7 @@ def require_resource(options = {})
   # This is called when an unknown method is encountered
   # "bare" in a control file - outside of a control or describe block.
   def method_missing(method_name, *arguments, &block)
+    require 'inspec/plugin/v2'
```

Comment:
@jerryaldrichiii, can you explain why this location was chosen for the `require`? It would seem more typical to require in the top of `lib/inspec/dsl.rb`, or in `lib/inspec/profile_context.rb` (which is where the error appeared).

Reply:
Great call out. I could use some words on describing the behavior succinctly. I believe the issue is because we are doing an `instance_eval`. The only places I could get it to work were to place the require here: https://github.com/inspec/inspec/blob/master/lib/inspec/profile_context.rb#L152 or the place you see it. I did the latter since it is closer to https://github.com/inspec/inspec/blob/852085969fe85d7a69157eb4304e075625ca8e68/lib/inspec/dsl.rb#L36

Placing at the top of the files like you would expect leads me down a rabbit hole that I couldn't get to the bottom of (issues like not being able to find `Inspec::BaseCLI`).

Reply:
If you'd like to pair or group work on this, I'd be happy to do so.

Reply:
Oh? Great! I'll bring it up in standup and we can get some time scheduled if the usual parking lot time doesn't cover it.